### PR TITLE
Fix warning about --no-hardcoded-sync being printed when it shouldn't

### PR DESCRIPTION
**Pull request by tomaka**
cc @tomusdrw

### CODE REVIEW

File: parity/run.rs [hunk 1]
```diff
@@ -575,7 +575,7 @@ pub fn execute_impl(cmd: RunCmd, can_restart: bool, logger: Arc<RotatingLogger>)
 	}
 
 	// display warning if using --no-hardcoded-sync
-	if !cmd.no_hardcoded_sync {
+	if cmd.no_hardcoded_sync {
```
Comment (5chdn) [inline parity/run.rs]: does this show this warning if you actually run --light?

Comment (tomusdrw) [reply parity/run.rs]: Yeah, this is in a function that sets up the full client only.

Comment (5chdn) [reply parity/run.rs]: Cool

Comment (tomaka) [reply parity/run.rs]: No, the `execute_impl` function is run only when we start a full node. This `if` is never reached on the light client.

**Pull request merged**